tests/acceptance/01_vars/02_functions/string.cf

```diff
@@ -0,0 +1,41 @@
+body common control
+{
+	bundlesequence => { "init", "check" };
+}
+
+bundle agent init
+{
+	classes:
+		"ExistingClass";
+
+	vars:
+		"a_int" int => "1k";
+		"a_real" real => "6.066";
+		"a_string" string => "Really interesting string";
+
+		"string_of_int" string => string("$(a_int)");
+		"string_of_real" string => string("$(a_real)");
+		"string_of_string" string => string("$(a_string)");
+		"string_of_bool__true" string => string(and("ExistingClass"));
+		"string_of_bool__false" string => string(and("NonExistingClass"));
+		"string_of_randint" string => string(randomint(10, 1000));
+
+#		"test_args" string => string();  # This is caught by the parser
+}
+
+bundle agent check
+{
+	classes:
+		"ok_int" expression => strcmp("$(init.a_int)", "$(init.string_of_int)");
+		"ok_real" expression => strcmp("$(init.a_real)", "$(init.string_of_real)");
+		"ok_string" expression => strcmp("$(init.a_string)", "$(init.string_of_string)");
```

Comment:
I think this is better:

```suggestion
		"ok_int" expression => strcmp("1000", "$(init.string_of_int)");
		"ok_real" expression => strcmp("6.066", "$(init.string_of_real)");
		"ok_string" expression => strcmp("Really interesting string", "$(init.string_of_string)");
```

I see 2 advantages:

- Test is more straight forward / easy to read
- What if both strings expand to `""` (empty string)? Your test would pass. (I guess this could happen if there is something wrong with variable expansion, or maybe you typed the wrong variable name).

In general, it is good to not repeat yourself, but a test is one of the exceptions where it makes sense.